api/configserver/save.go

```diff
@@ -145,6 +168,71 @@ func (s *Server) SaveConfig(w http.ResponseWriter, r *http.Request) {
 	s.writeSaveConfigResponse(w, SaveConfigResponse{Warnings: warnings}, session)
 }
 
+// Simply validate that the credentials exist; don't do anything with the actual secrets
+func validateCredParams(vars creds.Variables, config atc.Config, session lager.Logger) []string {
+	// 1. currently doesn't support tasks specified using a task file
+	// 2. should we check each of these are not nil first before Evaluating them to prevent unnecessary lookups, or is this already cached in vars?
+
+	errs := []string{}
+
+	for _, resourceType := range config.ResourceTypes {
+		_, err := creds.NewSource(vars, resourceType.Source).Evaluate()
+		if err != nil {
+			session.Debug(err.Error())
+			errs = append(errs, err.Error())
+		}
+	}
+
+	for _, resource := range config.Resources {
+		_, err := creds.NewSource(vars, resource.Source).Evaluate()
+		if err != nil {
+			session.Debug(err.Error())
+			errs = append(errs, err.Error())
+		}
+
+		_, err = creds.NewString(vars, resource.WebhookToken).Evaluate()
+		if err != nil {
+			session.Debug(err.Error())
+			errs = append(errs, err.Error())
+		}
+	}
+
+	for _, job := range config.Jobs {
+		for _, plan := range job.Plan {
+			_, err := creds.NewParams(vars, plan.Params).Evaluate()
+			if err != nil {
+				session.Debug(err.Error())
+				errs = append(errs, err.Error())
+			}
+
+			if plan.TaskConfig != nil {
+				if plan.TaskConfig.ImageResource != nil {
+					_, err = creds.NewSource(vars, plan.TaskConfig.ImageResource.Source).Evaluate()
+					if err != nil {
+						session.Debug(err.Error())
+						errs = append(errs, err.Error())
+					}
+				}
+
+				_, err = creds.NewTaskParams(vars, plan.TaskConfig.Params).Evaluate()
+				if err != nil {
+					session.Debug(err.Error())
+					errs = append(errs, err.Error())
+				}
+
+				// Are ImageResource.Params needed too?  According to the docs at https://concourse-ci.org/creds.html#vault,
+				// it seems like they aren't supported at the moment
+				// _, err = creds.NewParams(vars, plan.TaskConfig.ImageResource.Params).Evaluate()
```

Comment:
Yeah, I don't think we support evaluating `image_resource.params`. Support for `params` on `image_resource` was added for non-credentialy things. I'd expect all creds to be in `source`.